pkg/nats/stream/producer_client.go

```diff
@@ -143,7 +146,10 @@ func (pc *ProducerClient) heartBeat(ctx context.Context) {
 					continue
 				}
 
-				nonActiveStreamIds[c] = getStringList(heartBeatResponse.NonActiveStreamIds)
+				nonActiveStreamIdsFromConsumer := getStringList(heartBeatResponse.NonActiveStreamIds)
+				if len(nonActiveStreamIdsFromConsumer) != 0 {
+					nonActiveStreamIds[c] = append(nonActiveStreamIds[c], nonActiveStreamIdsFromConsumer...)
+				}
```

Comment:
why do we expect `nonActiveStreamIds` to have values for the same consumer from a previous loop iteration?

Reply:
Not sure I understand. This is still happening in the same loop right ! We are just flattening the map. The loop is for different consumers!

Reply:
What I mean is `nonActiveStreamIds[c] = append(nonActiveStreamIds[c], nonActiveStreamIdsFromConsumer...)` effectively the same as `nonActiveStreamIds[c] = nonActiveStreamIdsFromConsumer` because `nonActiveStreamIds[c]` is only populated once